src/ui/components/UploadScreen/Sharing.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { Dispatch, SetStateAction, useState } from "react";
+import hooks from "ui/hooks";
+import TeamSelect from "./TeamSelect";
+import { Workspace } from "ui/types";
+import { Toggle } from "../shared/Forms";
+import SettingsPreview from "./SettingsPreview";
+
+type SharingProps = {
+  workspaces: Workspace[];
+  selectedWorkspaceId: string | null;
+  setSelectedWorkspaceId: Dispatch<SetStateAction<string | null>>;
+  isPublic: boolean;
+  setIsPublic: Dispatch<SetStateAction<boolean>>;
+};
+
+function EditableSettings({
+  workspaces,
+  selectedWorkspaceId,
+  setSelectedWorkspaceId,
+  isPublic,
+  setIsPublic,
+}: Omit<SharingProps, "showSharingSettings">) {
+  const updateDefaultWorkspace = hooks.useUpdateDefaultWorkspace();
+  const handleWorkspaceSelect = (id: string | null) => {
```

Comment:
Is this null only because of the `My Library` case? In that case, a null-value object like `DEFAULT_WORKSPACE` might be better. Maybe watch https://www.youtube.com/watch?v=OMPfEXIlTVE&t=1526s if you've never seen it? Sandi Metz is 💯